project/design-patterns/src/prototype/Client.java

```
@@ -1,11 +1,15 @@
 package prototype;
 
 public class Client {
+	private static final String SHAPE_ID = "1";
```

I would use the CommandLine arguments to pass an id in and see what happens for different scenarios. Of course if you do so and decide to implement my suggestion of using the enum type as a key for your map, you'll also need to implement the parse method for the enum, as you'll need to convert the argument String to a ShapeType enum. (and in that case you'll probably want to throw an InvalidArgumentException if the argument doesn't match)